spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2475,10 +2478,14 @@ private void internalHeaders(final ConsumerRecord<K, V> record) {
 				record.headers().add(new RecordHeader(KafkaHeaders.DELIVERY_ATTEMPT, buff));
 			}
 			if (this.listenerinfo != null) {
-				record.headers().add(this.infoHeader);
+				listenerInfo(record);
 			}
 		}
 
+		private void listenerInfo(final ConsumerRecord<K, V> record) {
+			record.headers().add(this.infoHeader);
```

Comment:
Looks like we add the same info to all the records.
What is the reason of that `@Header(KafkaHeaders.LISTENER_INFO) List<String> infos`?
Cannot it be just one single header for the message based on a batch?
I don't mind having it in every single record of the batch we produce, but I doubt in a list of the same info...